CMakeLists.txt

```diff
@@ -12,53 +12,59 @@ target_include_directories(clap-core INTERFACE include)
 
 install(DIRECTORY include DESTINATION "." OPTIONAL EXCLUDE_FROM_ALL)
 
-add_executable(clap-compile-c EXCLUDE_FROM_ALL src/main.c)
+if (${CLAP_BUILD_TESTS})
+    message(STATUS "Including clap tests, compile tests, and versions")
+    add_custom_target(clap-tests)
 
-macro(clap_compile_cpp SUFFIX EXT STDC STDCPP)
-    add_executable(clap-compile-${SUFFIX} EXCLUDE_FROM_ALL src/main.${EXT})
-    target_link_libraries(clap-compile-${SUFFIX} clap-core)
-    set_target_properties(clap-compile-${SUFFIX} PROPERTIES
-        C_STANDARD ${STDC}
-        CXX_STANDARD ${STDCPP})
-    add_test(NAME test-clap-compile-${SUFFIX} COMMAND clap-compile-${SUFFIX})
-    add_dependencies(clap-tests clap-compile-${SUFFIX})
+    add_executable(clap-compile-c EXCLUDE_FROM_ALL src/main.c)
```

Comment:
Isn't this redundant? As this would compile the example with the compiler's default settings, but we mostly care about the targets that are specifically tested. Also, this is not built on the CI so this target is never actually used anywhere.